staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go

```diff
@@ -64,7 +64,35 @@ func StorageWithCacher(capacity int) generic.StorageDecorator {
 			cacher.Stop()
 			d()
 		}
-
+		RegisterStorageCleanup(destroyFunc)
 		return cacher, destroyFunc
 	}
 }
+
+// HACK ALERT : Track the destroy methods to call them
+// from the test harness. TrackStorageCleanup will be called
+// only from the test harness, so Register/Cleanup will be
+// no-op at runtime.
```

Comment:
TODO: remove this when we shutdown storages correctly, <link to @frobware's PR>

Reply:
Done